spec/services/conviction_expiry_calculator_spec.rb

```diff
@@ -1,14 +1,38 @@
 require 'rails_helper'
 
 RSpec.describe ConvictionExpiryCalculator do
-  subject { described_class.new(disclosure_check: disclosure_check) }
+  subject { described_class.new(disclosure_check: disclosure_check, ) }
 
   context '#expiry_date' do
-    let(:disclosure_check) { build(:disclosure_check, :conviction) }
+    let(:disclosure_check) { build(:disclosure_check, :conviction,
+                                   conviction_subtype: conviction_subtype,
+                                   known_date: known_date,
+                                   conviction_length: conviction_length,
+                                   conviction_length_type: conviction_length_type) }
+
     let(:expiry_date) { subject.expiry_date }
+    let(:known_date) { nil }
```

Comment:
Maybe we can try leaving the operations out of this spec, as otherwise we will end up having very similar or duplicated tests, the ones here, and the ones in each of the calculators.

The ideal would be to thoroughly test each of the individual 'calculators', and then ensure in this spec we call the proper calculator, but not really care about the results as the individual calculators should have test already that.